### PR TITLE
Add Testing Scripts

### PR DESCRIPTION
**Pull request by jonmchan**
## what
* Add Test Scripts

## why
* Provides us the ability to add some level of CI testing for PRs to ensure that at minimum Google Authenticator, SSH, sudosh setup, and HTTPS connection to slack API continues to work.

## references
* Adopted from neochrome/docker-bastion - https://github.com/neochrome/docker-bastion/tree/master/test - thank you @neochrome!

Depends on #70

### CODE REVIEW

File: test/ida_rsa [hunk 1]
```diff
@@ -0,0 +1,38 @@
+-----BEGIN OPENSSH PRIVATE KEY-----
```
Comment (max-lobur) [inline test/ida_rsa:1]: Once in a while this will fire in various security scanners and reports. I would rather generate the key on the fly in a test to avoid the noise

File: test/test.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+if [ $retVal -ne 0 ]; then
+  echo "* sudosh Audit Failed - no logs created!"
+  exit $retVal
+else
+  echo "* sudosh Audit Test Succeeded"
+fi
+
+
+docker compose exec bastion curl https://hooks.slack.com
```
Comment (max-lobur) [inline test/test.sh:32]: What was the test case for this? resolv.conf broken or something?

File: test/ida_rsa (hunk 1 above)
Comment (osterman) [reply test/ida_rsa:1]: That would be ideal

File: test/test.sh (hunk 2 above)
Comment (jonmchan) [reply test/test.sh:32]: https://github.com/cloudposse/bastion/issues/67 - this is to catch the Root CA certs being out of date issue and validate/verify that the slack hook functionality is not broken.

File: test/ida_rsa (hunk 1 above)
Comment (jonmchan) [reply test/ida_rsa]: Made this dynamically generated. The test runner will have to have the `ssh-keygen` command available.

**Review by max-lobur — DISMISSED** [review]

**Review by max-lobur — APPROVED** [review]
